CRM/Xdedupe/Resolver/CustomField.php

```diff
@@ -0,0 +1,188 @@
+| at www.gnu.org/licenses/agpl.html. Removal of this     |
+| copyright header is strictly prohibited without        |
+| written permission from the original author(s).        |
++--------------------------------------------------------*/
+
+use CRM_Xdedupe_ExtensionUtil as E;
+
+/**
+ * Implements a resolver for contact custom fields
+ */
+class CRM_Xdedupe_Resolver_CustomField extends CRM_Xdedupe_Resolver
+{
+
+    /** @var integer ID of the custom field */
+    protected $custom_field_id = null;
+
+    public function __construct($merge, $custom_field_id)
+    {
+        $this->custom_field_id = $custom_field_id;
+        parent::__construct($merge);
+    }
+
+    /**
+     * Get the spec (i.e. class name) that refers to this resolver
+     * @return string spec string
+     */
+    public function getSpec()
+    {
+        return "CRM_Xdedupe_Resolver_CustomField:{$this->custom_field_id}";
+    }
+
+    /**
+     * Report the contact attributes that this resolver requires
+     *
+     * @return array list of contact attributes
+     */
+    public function getContactAttributes()
+    {
+        return ["custom_{$this->custom_field_id}"];
+    }
+
+    /**
+     * get the name of the finder
+     * @return string name
+     */
+    public function getName()
+    {
+        // $field_name = civicrm_api3('CustomField', 'getvalue', ['id' => $this->custom_field_id, 'return' => 'label']);
+        $field_name = civicrm_api4('CustomField', 'get', [ 'select' => [ 'label','custom_group_id',], 'where' => [ [ 'id', '=', $this->custom_field_id ] , ], ]);
+        $group_name = civicrm_api4('CustomGroup', 'get', [ 'select' => [ 'title',], 'where' => [ [ 'id', '=', $field_name[0]['custom_group_id'] ] , ], ]);
+        //\Civi::log('xdedupe')->debug('xdedupe: customfields {original}', ['original' => $group_name,]);
+        return E::ts("'%1' Custom Field of Group '%2'", [1 => $field_name[0]['label'], 2 =>  $group_name[0]['title']]);
+    }
+
+    /**
+     * get an explanation what the finder does
+     * @return string name
+     */
+    public function getHelp()
+    {
+        $field_name = civicrm_api4('CustomField', 'get', [ 'select' => [ 'label',], 'where' => [ [ 'id', '=', $this->custom_field_id ] , ], ]);
+        return E::ts(
+            "The field '%1' is a custom field. This resolver will merge the values of all duplicates, so that the main contact will have all.",
+            [1 => $field_name[0]['label']]
+        );
+    }
+
+    /**
+     * Get the contact's field value
+     *
+     * @param $contact_id integer contact ID
+     * @return mixed
+     */
+    protected function getValue($contact_id)
+    {
+        $field_name = "custom_{$this->custom_field_id}";
+        $contact = $this->getContext()->getContact($contact_id);
+//        Civi::log()->debug("Field name is {$field_name}, contact data is: " . json_encode($contact));
+        return CRM_Utils_Array::value($field_name, $contact, '');
```

Comment:
`CRM_Utils_Array::value()` is deprecated and should be replaced by `$contact[$field_name] ?? ''`.